File: Ahorn/entities/doorField.jl
module CH_DoorField

using ..Ahorn, Maple
sides=String[
    "A",
    "B",
    "C"]
@mapdef Entity "CherryHelper/DoorField" DoorField(x::Integer, y::Integer, width::Integer=8, height::Integer=8,  toLevel::String="",EndLevel::Bool=false, SIDOfMap::String="", LoadAnotherBin::Bool=false, Side::String="A")

const placements = Ahorn.PlacementDict(
    "Door Field (Room Teleport) (Cherry Helper)" => Ahorn.EntityPlacement(
         DoorField,
         "rectangle",
         Dict{String, Any}(
            "EndLevel" => false,
            "LoadAnotherBin" => false
        )
    ),
    "Door Field (End Level) (Cherry Helper)" => Ahorn.EntityPlacement(
         DoorField,
         "rectangle",
         Dict{String, Any}(
            "EndLevel" => true,
            "LoadAnotherBin" => false
        )
    ),
    "Door Field (Chapter Change) (Cherry Helper)" => Ahorn.EntityPlacement(
         DoorField,
         "rectangle",
         Dict{String, Any}(
            "EndLevel" => false,
            "LoadAnotherBin" => true,
            "SIDOfMap" => "Celeste/1-ForsakenCity",
            "Side" => "A"
        )
    )
)

Ahorn.minimumSize(entity::DoorField) = 8, 8
Ahorn.resizable(entity::DoorField) = true, true


Ahorn.editingOptions(entity::DoorField) = Dict{String, Any}(
    "Side" => sides
)
Ahorn.selection(entity::DoorField) = Ahorn.getEntityRectangle(entity)

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::DoorField, room::Maple.Room)
    width = Int(get(entity.data, "width", 32))
    height = Int(get(entity.data, "height", 32))

    Ahorn.drawRectangle(ctx, 0, 0, width, height, (0.0, 0.0, 0.0, 1.0), (0.2, 0.2, 0.6, 1.0))
end

end

File: Loenn/entities/doorField.lua
local doorField = {}

local sides = {
    A = "A",
    B = "B",
    C = "C"
}

doorField.name = "CherryHelper/DoorField"
doorField.justification ={0.5, 0.5}
doorField.borderColor = {0.2, 0.2, 0.6, 1.0}
doorField.fillColor = {0.0, 0.0, 0.0, 1.0}
doorField.fieldInformation = {
    Side = {
        options = sides,
        editable = false
    }
}
doorField.placements = {
    {
        name = "roomTeleport",
        data = {
            toLevel="",
            EndLevel=false,
            SIDOfMap="",
            LoadAnotherBin=false,
            Side="A",
            width=16,
            height=16,
            spawnOffsetX=0,
            spawnOffsetY=0
        }
    },
    {
        name = "endLevel",
        data = {
            toLevel="",
            EndLevel=true,
            SIDOfMap="",
            LoadAnotherBin=false,
            Side="A",
            width=16,
            height=16,
            spawnOffsetX=0,
            spawnOffsetY=0
        }
    },
    {
        name = "chapterChange",
        data = {
            toLevel="",
            EndLevel=false,
            SIDOfMap="Celeste/1-ForsakenCity",
            LoadAnotherBin=true,
            Side="A",
            width=16,
            height=16,
            spawnOffsetX=0,
            spawnOffsetY=0
        }
    }
}


return doorField